detector.go

```diff
@@ -205,9 +207,10 @@ func (d *Detector) detectGroup(group buildpack.Group, done []buildpack.GroupElem
 		go func(key string, bp buildpack.BuildModule) {
 			if _, ok := d.Runs.Load(key); !ok {
 				detectConfig := &buildpack.DetectConfig{
-					AppDir:      d.AppDir,
-					PlatformDir: d.PlatformDir,
-					Logger:      d.Logger,
+					AppDir:         d.AppDir,
+					PlatformDir:    d.PlatformDir,
+					BuildConfigDir: d.BuildConfigDir,
```

Comment:
Similar comment to https://github.com/buildpacks/lifecycle/pull/899/files#r1021721179, here is where we do it for the detector: https://github.com/buildpacks/lifecycle/blob/ef9a1449e43973f7e147cdd54689686b4816dffa/detector_test.go#L79